unittest/src/fake/leds/mod.rs

```diff
@@ -0,0 +1,87 @@
+//! Fake implementation of the LEDs API, documented here:
+//! https://github.com/tock/tock/blob/master/doc/syscalls/00002_leds.md
+//!
+//! Like the real API, `Leds` controls a set of fake LEDs. It provides
+//! a function `get_led` used to retrieve the state of an LED.
+
+use core::cell::Cell;
+use libtock_platform::{CommandReturn, ErrorCode};
+
+pub struct Leds<const LEDS_COUNT: usize> {
+    leds: [Cell<bool>; LEDS_COUNT],
+}
+
+impl<const LEDS_COUNT: usize> Leds<LEDS_COUNT> {
+    pub fn new() -> std::rc::Rc<Leds<LEDS_COUNT>> {
+        #[allow(clippy::declare_interior_mutable_const)]
+        const OFF: Cell<bool> = Cell::new(false);
+        std::rc::Rc::new(Leds {
+            leds: [OFF; LEDS_COUNT],
```

Comment:
You can use `Default::default()` to initialize `leds` and avoid the clippy warning entirely.

Reply:
Doesn't seem to work. Using `[Default::default(); LEDS_COUNT]` says that `Cell<bool>` does not implement `Copy`, while using `const OFF: Cell<bool> = Default::default()` says that `default` is not a `const fn`. I might be missing something though.

Reply:
I tried this:

```rust
const fn default() -> Cell<bool> {
    Cell::new(false)
}

std::rc::Rc::new(Leds {
    leds: [default(); LEDS_COUNT],
})
```

The compiler still states that `Copy` is not implemented.

Writing 

```rust
const fn default() -> Cell<bool> {
    Cell::new(false)
}
const OFF: Cell<bool> = default();
std::rc::Rc::new(Leds {
    leds: [OFF; LEDS_COUNT],
})
```

makes clippy show the warning.

Reply:
https://github.com/rust-lang/rust-clippy/issues/7665 based on this discussion it looks like the recommended approach is just to ignore the lint, sorry for the noise